### PR TITLE
TLS-Memoryleak

### LINKED ISSUE #2278: FleetAutoscaler keeps alive all TLS connections permanently causing memory leak on webhook server
**What happened**:
Over time my `https` server which hosts the FleetAutoscaler webhook goes OOM. This is caused by 1000s of never dying sockets on the server. This does NOT happen when I call it with cURL or a browser. It only happens with Agones calls the endpoint.

```
/app $ lsof -p $PID | grep socket
...
1       /app/zeus-rest  socket:[289294771]
1       /app/zeus-rest  socket:[289294783]
1       /app/zeus-rest  socket:[289292336]
1       /app/zeus-rest  socket:[289291653]
1       /app/zeus-rest  socket:[289291654]
1       /app/zeus-rest  socket:[289293769]
1       /app/zeus-rest  socket:[289294780]
...

/app $ lsof -p $PID | grep socket | wc -l
6397
/app $ lsof -p $PID | grep socket | wc -l
6403
/app $ lsof -p $PID | grep socket | wc -l
6418
```

**What you expected to happen**:

I expect that when the FleetAutoscaler is called by Agones is either reuses the TLS client or it disconnects it. Keeping it alive and then making a new one seems naughty.

**How to reproduce it (as minimally and precisely as possible)**:
Create a TLS FleetAutoscaler endpoint with keepalive turned on and no timeout specified and watch the sockets multiply.

**Anything else we need to know?**:
I suspect that this could be repaired by adding to 
`pkg/fleetautoscalers/fleetautoscalers.go`
```
var client = http.Client{
	Timeout: 15 * time.Second,
+++	Transport: &http.Transport{
+++                DisableKeepAlives: true,
+++        },
}
```
I fixed it by disabling KeepAlive on the server side. But it took me several hours to figure out the problem because I could not reproduce it with any clients of my own.

**Environment**:
- Agones version: 1.16
- Kubernetes version (use `kubectl version`): 1.21
- Cloud provider or hardware configuration: EKS and Minikube
- Install method (yaml/helm): helm
- Troubleshooting guide log(s):
- Others:

### PR DESCRIPTION
**Pull request by SaitejaTamma**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, press enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking

 /kind bug

> /kind cleanup
> /kind documentation
> /kind feature
> /kind hotfix

**What this PR does / Why we need it**:

**Which issue(s) this PR fixes**:

Closes #2278 

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/fleetautoscalers/fleetautoscalers.go [hunk 1]
```diff
@@ -115,11 +115,8 @@ func setCABundle(caBundle []byte) error {
 	if ok := rootCAs.AppendCertsFromPEM(caBundle); !ok {
 		return errors.New("no certs were appended from caBundle")
 	}
-	client.Transport = &http.Transport{
-		TLSClientConfig: &tls.Config{
-			RootCAs: rootCAs,
-		},
-	}
+	TLSClientConfig := &tls.Config{}
```
Comment (markmandel) [inline pkg/fleetautoscalers/fleetautoscalers.go:118]: What we need to do is update this `client` global variable here:
https://github.com/googleforgames/agones/blob/0ec68ff6953e30868615f3ca333876bce7f3b894/pkg/fleetautoscalers/fleetautoscalers.go#L38

What is happening here is you are changing a local variable now, which isn't doing anything specific.

File: pkg/fleetautoscalers/fleetautoscalers.go [hunk 2]
```diff
@@ -115,11 +119,7 @@ func setCABundle(caBundle []byte) error {
 	if ok := rootCAs.AppendCertsFromPEM(caBundle); !ok {
 		return errors.New("no certs were appended from caBundle")
 	}
-	client.Transport = &http.Transport{
-		TLSClientConfig: &tls.Config{
-			RootCAs: rootCAs,
-		},
-	}
+	tlsConfig.RootCAs = rootCAs
```
Comment (markmandel) [inline pkg/fleetautoscalers/fleetautoscalers.go:122]: 🤔 

Does it still work if we do `client.Transport.TLSCClientConfig.RootCAs = rootCAs` instead? (then we could inline `tlsConfig` into `client` in the `var` declaration above?

Comment (SaitejaTamma) [reply pkg/fleetautoscalers/fleetautoscalers.go:122]: I tried this approach as well but instance is not created by default, TLSclientconfig do not present when we assign align like that. By default it is having the round trip instance not the transport instance.

Comment (markmandel) [reply pkg/fleetautoscalers/fleetautoscalers.go:122]: What if above becomes:

```go
var client = http.Client{
	Timeout: 15 * time.Second,
	Transport: &http.Transport{
		TLSClientConfig: &tls.Config{},
	},
}
```

Does that resolve the issue?

Comment (SaitejaTamma) [reply pkg/fleetautoscalers/fleetautoscalers.go:122]: If we  dont have a tlsConfig variable we will not be able to reference it to update the rootCAs at https://github.com/googleforgames/agones/blob/main/pkg/fleetautoscalers/fleetautoscalers.go#L118-L121.

Comment (markmandel) [reply pkg/fleetautoscalers/fleetautoscalers.go:122]: Oh I see the issue now, `http.RoundTripper` is an interface, so you can't access the variables within it.

I was playing with it here: https://go.dev/play/p/C6RTLx9VfWl

**Review by markmandel — APPROVED** [review]

**Pull request merged**